packages/kbn-apm-synthtrace/src/lib/apm/instance.ts

```diff
@@ -14,15 +14,31 @@ import { Transaction } from './transaction';
 import { ApmApplicationMetricFields, ApmFields } from './apm_fields';
 
 export class Instance extends Entity<ApmFields> {
-  transaction(transactionName: string, transactionType = 'request') {
+  transaction({
+    transactionName,
+    transactionType = 'request',
+  }: {
+    transactionName: string;
+    transactionType?: string;
+  }) {
     return new Transaction({
       ...this.fields,
       'transaction.name': transactionName,
       'transaction.type': transactionType,
     });
   }
 
-  span(spanName: string, spanType: string, spanSubtype?: string, apmFields?: ApmFields) {
+  span({
+    spanName,
+    spanType,
+    spanSubtype,
+    apmFields,
+  }: {
+    spanName: string;
+    spanType: string;
+    spanSubtype?: string;
+    apmFields?: ApmFields;
+  }) {
```

Comment:
We might want to destructure `apmFields` so the signature becomes something like:

```suggestion
  span({
    spanName,
    spanType,
    spanSubtype,
    ...apmFields,
  }: {
    spanName: string;
    spanType: string;
    spanSubtype?: string;
  } & ApmFields) {
```

Meaning we can do:
```
span({ spanName: 'foo', spanType: 'bar', 'kubernetes.pod.name': 'baz' })
```